### PR TITLE
Demo One Password input

### PR DESCRIPTION
**Pull request by nitro-marky**
This takes the user password and stores it in @n3op2 's custom context to be then passes to the rest of the system.

Screenshot 2022-08-15 at 11 39 24

Screenshot 2022-08-15 at 11 39 34

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
 export default function App() {
   return (
     <Root display={'flex'} justifyContent={'center'}>
       <Container
-        size={10}
         styles={{
-          flexDirection: 'column',
-          maxHeight: '1024px',
-          maxWidth: '1366px',
+          maxHeight: '1080px',
```
Comment (n3op2) [inline src/components/App.js:10]: The screen size I've taken from figma. Not sure if this was updated or not.

File: src/components/Card.js [hunk 2]
```diff
@@ -20,8 +20,8 @@ const Paper = styled.div`
   align-items: flex-start;
   text-align: left;
   opacity: 0.6;
-  width: 317px;
-  height: 355px;
+  width: 400px;
```
Comment (n3op2) [inline src/components/Card.js:23]: Why this has been changed? I taken card sizes from figma

File: src/components/Demo1/DemoOneBackground.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components'
+
+import backgroundDemoOne from '../../assets/images/background-demo-one.png'
+
+export const DemoOneBackground = styled.div(() => ({
```
Comment (n3op2) [inline src/components/Demo1/DemoOneBackground.js:5]: Why all files got prefixes of `DemoOne` when it's in a folder `Demo1`? 

Can we call just `Background` and 'Header' and so on?

File: src/components/Demo1/DemoOneBox.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+const Body = styled.div({
+  boxSizing: 'border-box',
+  height: '100%',
+  width: '100%',
+  padding: '10px',
+  outline: '2px solid #FFFFFF',
+  outlineOffset: '-10px',
+})
+
+export default function DemoOneBox(props) {
```
Comment (n3op2) [inline src/components/Demo1/DemoOneBox.js:67]: It could be just `Box`

File: src/components/MainMenu.js [hunk 5]
```diff
@@ -21,20 +21,27 @@ export default function MainMenu() {
   return (
     <Container
       size={10}
-      style={{ justifyContent: 'flex-start', flexFlow: 'row wrap' }}
+      style={{ flexDirection: 'column', justifyContent: 'center' }}
```
Comment (n3op2) [inline src/components/MainMenu.js:24]: Why main menu was changed? Also layout is wrong.  Please do not make changes as part of this PR as I already started addressing

File: src/layout/Demo1.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+`
+
+export default function DemoOne(props) {
```
Comment (n3op2) [inline src/layout/Demo1.js:18]: Again please rename component without prefix of `DemoOne`. It's in a folder `Demo1`.

**Review by n3op2 — CHANGES_REQUESTED** [review]

File: src/components/App.js (hunk 1 above)
Comment (nitro-marky) [reply src/components/App.js:10]: We were told that the resolution was going to be 1920x1080, so Jon set the background to that size

File: src/components/MainMenu.js (hunk 5 above)
Comment (nitro-marky) [reply src/components/MainMenu.js:24]: That again was not my doing

File: src/layout/Demo1.js (hunk 6 above)
Comment (nitro-marky) [reply src/layout/Demo1.js:18]: Wouldn't it be easier to change the name of the folder??

File: src/components/MainMenu.js (hunk 5 above)
Comment (n3op2) [reply src/components/MainMenu.js:24]: so then why do you include in your PR?

File: src/layout/Demo1.js (hunk 6 above)
Comment (n3op2) [reply src/layout/Demo1.js:18]: No, leave folder as itt's and update your components, we will keep that folder and add more for Demo2, Demo3 and so on.

File: src/components/MainMenu.js (hunk 5 above)
Comment (nitro-marky) [reply src/components/MainMenu.js:24]: Because Jon did some stuff on my branch

File: src/fixtures/demos.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { Navigate } from 'react-router-dom'
 
-import DemoOne from '../layout/Demo1'
+import Demo from '../layout/Demo1'
```
Comment (n3op2) [inline src/fixtures/demos.js:4]: Please move this inside `../components/Demo1` folder

File: src/components/Demo1/Box.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+              <Text>
+                This application will store your password securely.
+                <br />
+                Please input a keyword of choice.
+              </Text>
+              <Input />
+            </>
+          )}
+          {isPasswordSet && (
+            <>
```
Comment (n3op2) [inline src/components/Demo1/Box.js:94]: Why do we use React.Fragment here? It's already wrapped in <SavedText>?

File: src/components/Demo1/Index.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
```
Comment (n3op2) [inline src/components/Demo1/Index.js:1]: please rename `Index.js` to `index.js`, lower case `I`

File: src/fixtures/demos.js [hunk 10]
```diff
@@ -1,6 +1,8 @@
 import React from 'react'
 import { Navigate } from 'react-router-dom'
 
+import Demo1 from '../../src/components/Demo1/Index'
```
Comment (n3op2) [inline src/fixtures/demos.js:4]: Again, please rename this to index so in this import you can use

`import Demo1 from '../../src/components/Demo1'`

**Review by n3op2 — APPROVED** [review]
Approving but there are a lot of places where this could be improved:
- reusable and configurable components
- quite a few styled components could be abstracted
- more coverage in integration tests
- handling of 16 chars could render an error message
- sanitize password?
- button coould be disabled/enabled depending on user's inputt e.g. less than 3 chars could render it disabled and more than 16chars could also render it disabled along with error message.

**Pull request merged**